test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -0,0 +1,328 @@
+        });
+
+        test('with two or more fields', () => {
+            query = `Query { scalars { num } test { name } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(3); // Query 1 + scalars 1 + test 1
+        });
+
+        test('with one level of nested fields', () => {
+            query = `Query { scalars { num, test { name } } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(3); // Query 1 + scalars 1 + test 1
+        });
+
+        test('with multiple levels of nesting', () => {
+            query = `Query { scalars { num, test { name, scalars { id } } } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(4); // Query 1 + scalars 1 + test 1 + scalars 1
+        });
+
+        test('with aliases', () => {
+            query = `Query { foo: scalar { num } bar: scalar { id }}`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(3); // Query 1 + scalar 1 + scalar 1
+        });
+
+        test('with all scalar fields', () => {
+            query = `Query { scalars { id, num, float, bool, string } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(2); // Query 1 + scalar 1
+        });
+
+        test('with arguments and variables', () => {
+            query = `Query { hero(episode: EMPIRE) { id, name } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(2); // Query 1 + hero/character 1
+            query = `Query { human(id: 1) { id, name, appearsIn } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(3); // Query 1 + human/character 1 + appearsIn/episode
+            // argument passed in as a variable
+            query = `Query { hero(episode: $ep) { id, name } }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(2); // Query 1 + hero/character 1
+        });
+
+        test('with fragments', () => {
+            query = `
+            Query {
+                leftComparison: hero(episode: EMPIRE) {
+                  ...comparisonFields
+                }
+                rightComparison: hero(episode: JEDI) {
+                  ...comparisonFields
+                }
+              }
+              
+              fragment comparisonFields on Character {
+                name
+                appearsIn
+              }
+            }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(5); // Query 1 + 2*(character 1 + appearsIn/episode 1)
+        });
+
+        test('with inline fragments', () => {
+            query = `
+            Query {
+                hero(episode: EMPIRE) {
+                    name
+                    ... on Droid {
+                        primaryFunction
+                    }
+                    ... on Human {
+                        homeplanet
+                    }
+                }
+            }`;
+            expect(getQueryTypeComplxity(query, typeWeights)).toBe(2); // Query 1 + hero/character 1)
```

Comment:
Should the expected result be 3 since the Query, hero and Droid/Human resolvers would be called?

Reply:
The `hero` field returns the character interface type. `Droid` or `Human` are just extensions of the `Character `interface and I don't think they would require extra resolvers to determine. Regardless, this is type complexity so the response only has 1 object type in as a result of the `hero` field, so all that just ends up being 1. Thats how I'm thinking about it, we can discuss.